--- /usr/share/pve-manager/js/pvemanagerlib.js.bak
+++ /usr/share/pve-manager/js/pvemanagerlib.js
@@ -3454,6 +3454,17 @@
             }
         },
 
+        lxc_entry_count: 256,
+
+        forEachLxcEntry: function (func) {
+            for (let i = 0; i < PVE.Utils.lxc_entry_count; i++) {
+                let cont = func(i, `entry${i}`);
+                if (!cont && cont !== undefined) {
+                    return;
+                }
+            }
+        },
+
         hardware_counts: {
             net: 32,
             usb: 14,
@@ -40435,6 +40446,15 @@
         },
         {
             xtype: 'proxmoxcheckbox',
+            fieldLabel: gettext('AutoDev'),
+            name: 'autodev',
+            bind: {
+                disabled: '{unprivileged}',
+                boxLabel: '{privilegedOnly}',
+            },
+        },
+        {
+            xtype: 'proxmoxcheckbox',
             fieldLabel: gettext('Nesting'),
             name: 'nesting',
         },
@@ -40747,6 +40767,256 @@
         },
     ],
 });
+Ext.define('PVE.lxc.AutomountEdit', {
+    extend: 'Proxmox.window.Edit',
+    xtype: 'pveLxcAutomountEdit',
+
+    subject: gettext('lxc.mount.auto'),
+    autoLoad: true,
+    width: 350,
+
+    items: [{
+	xtype: 'pveLxcAutomountInputPanel',
+    }],
+});
+Ext.define('PVE.lxc.AutomountInputPanel', {
+    extend: 'Proxmox.panel.InputPanel',
+    xtype: 'pveLxcAutomountInputPanel',
+
+    mountOptions: ['proc', 'sys', 'cgroup'],
+
+    items: [
+        {
+            xtype: 'proxmoxKVComboBox',
+            name: 'proc',
+            fieldLabel: gettext('proc'),
+            deleteEmpty: true,
+            comboItems: [
+                ['None', 'None'],
+                ['ro', 'ro'],
+                ['mixed', 'mixed'],
+                ['rw', 'rw']
+            ],
+            value: 'None',
+        },
+        {
+            xtype: 'proxmoxKVComboBox',
+            name: 'sys',
+            fieldLabel: gettext('sys'),
+            deleteEmpty: true,
+            comboItems: [
+                ['None', 'None'],
+                ['ro', 'ro'],
+                ['mixed', 'mixed'],
+                ['rw', 'rw']
+            ],
+            value: 'None',
+        },
+        {
+            xtype: 'proxmoxKVComboBox',
+            name: 'cgroup',
+            fieldLabel: gettext('cgroup'),
+            deleteEmpty: true,
+            comboItems: [
+                ['None', 'None'],
+                ['ro', 'ro'],
+                ['mixed', 'mixed'],
+                ['rw', 'rw']
+            ],
+            value: 'None',
+        },
+    ],
+
+    onGetValues: function(values) {
+        var result = [];
+
+        this.mountOptions.forEach(function(option) {
+            if (values[option] && values[option] !== 'None') {
+                result.push(option + '=' + values[option]);
+            }
+            delete values[option]
+        });
+
+	if (result.length) {
+	    values.automount = result.join(',');
+	} else {
+	    values.delete = 'automount';
+	}
+
+        return values;
+    },
+
+    setValues: function(values) {
+        var me = this;
+
+        me.mountOptions.forEach(function(option) {
+            if (values.automount && values.automount.includes(option + '=')) {
+                values[option] = values.automount.split(',').find(function(item) {
+                    return item.startsWith(option + '=');
+                }).split('=')[1];
+            }
+        });
+
+        this.callParent([values]);
+    },
+
+    initComponent: function() {
+        let me = this;
+        me.callParent();
+    },
+});
+Ext.define('PVE.lxc.EntryEdit', {
+    extend: 'Proxmox.window.Edit',
+
+    vmconfig: undefined,
+
+    isAdd: true,
+    width: 450,
+
+    initComponent: function() {
+	let me = this;
+
+	me.isCreate = !me.confid;
+
+	let ipanel = Ext.create('PVE.lxc.EntryInputPanel', {
+	    confid: me.confid,
+	    isCreate: me.isCreate,
+	    pveSelNode: me.pveSelNode,
+	});
+
+	let subject;
+	if (me.isCreate) {
+	    subject = gettext('Entry');
+	} else {
+	    subject = gettext('Entry') + ' (' + me.confid + ')';
+	}
+
+	Ext.apply(me, {
+	    subject: subject,
+	    items: [ipanel],
+	});
+
+	me.callParent();
+
+	me.load({
+	    success: function(response, options) {
+		ipanel.setVMConfig(response.result.data);
+		if (me.isCreate) {
+		    return;
+		}
+
+		let data = PVE.Parser.parsePropertyString(response.result.data[me.confid], 'path');
+
+		let values = {
+		    path1: data.path1,
+		    path2: data.path2,
+		    create: data.create,
+		};
+
+		ipanel.setValues(values);
+	    },
+	});
+    },
+});
+Ext.define('PVE.lxc.EntryInputPanel', {
+    extend: 'Proxmox.panel.InputPanel',
+    mixins: ['Proxmox.Mixin.CBind'],
+
+    autoComplete: false,
+
+    controller: {
+	xclass: 'Ext.app.ViewController',
+    },
+
+    setVMConfig: function(vmconfig) {
+	let me = this;
+	me.vmconfig = vmconfig;
+
+	if (me.isCreate) {
+	    PVE.Utils.forEachLxcEntry((i, name) => {
+		if (!Ext.isDefined(vmconfig[name])) {
+		    me.confid = name;
+		    me.down('field[name=entryid]').setValue(i);
+		    return false;
+		}
+		return undefined;
+	    });
+	}
+    },
+
+    onGetValues: function(values) {
+	let me = this;
+	let confid = me.isCreate ? "entry" + values.entryid : me.confid;
+	delete values.entryid;
+	let val = PVE.Parser.printPropertyString(values, 'path');
+	let ret = {};
+	ret[confid] = val;
+	return ret;
+    },
+
+    items: [
+	{
+	    xtype: 'proxmoxintegerfield',
+	    name: 'entryid',
+	    minValue: 0,
+	    maxValue: 6,
+	    hidden: true,
+	    allowBlank: false,
+	    disabled: true,
+	    cbind: {
+		disabled: '{!isCreate}',
+	    },
+	},
+	{
+	    xtype: 'textfield',
+	    name: 'path1',
+	    fieldLabel: gettext('Source Path'),
+	    labelWidth: 120,
+	    editable: true,
+	    allowBlank: false,
+	    emptyText: '/dev/xyz',
+	    validator: function(v) {
+		if (!v.startsWith('/dev/')) {
+		    return gettext("Path must start with /dev/");
+		}
+		if (!/^\/dev\/[\w/-]+$/.test(v)) {
+		    return gettext("Invalid path format");
+		}
+		return true;
+	    },
+	},
+	{
+	    xtype: 'textfield',
+	    name: 'path2',
+	    fieldLabel: gettext('Target Path'),
+	    labelWidth: 120,
+	    editable: true,
+	    allowBlank: false,
+	    emptyText: '/dev/xyz',
+	    validator: function(v) {
+		if (!v.startsWith('/dev/')) {
+		    return gettext("Path must start with /dev/");
+		}
+		if (!/^\/dev\/[\w/-]+$/.test(v)) {
+		    return gettext("Invalid path format");
+		}
+		return true;
+	    },
+	},
+	{
+	    xtype: 'combo',
+	    name: 'create',
+	    fieldLabel: gettext('Create Type'),
+	    labelWidth: 120,
+	    editable: false,
+	    allowBlank: false,
+	    store: ['file', 'dir'],
+	    value: 'file',
+	    triggerAction: 'all',
+	    emptyText: gettext('Select create type'),
+	},
+    ],
+});
 Ext.define('PVE.lxc.IdMapField', {
     extend: 'Ext.form.FieldContainer',
     xtype: 'pveLxcIdMapField',
@@ -42291,6 +42561,11 @@
                 defaultValue: Proxmox.Utils.noneText,
                 editor: 'PVE.lxc.FeaturesEdit',
             },
+            automount: {
+                header: gettext('lxc.mount.auto'),
+                defaultValue: Proxmox.Utils.noneText,
+                editor: 'PVE.lxc.AutomountEdit',
+            },
             hookscript: {
                 header: gettext('Hookscript'),
                 renderer: Ext.htmlEncode,
@@ -42393,11 +42668,22 @@
 
         me.callParent();
 
+        me.store.addFilter(new Ext.util.Filter({
+            filterFn: function(item) {
+                let ostype = me.store.getById('ostype');
+                if (ostype && ostype.data.value !== 'oci') {
+                    return item.data.key !== 'automount';
+                }
+                return true;
+            }
+        }));
+
         me.on('activate', me.rstore.startUpdate);
         me.on('destroy', me.rstore.stopUpdate);
         me.on('deactivate', me.rstore.stopUpdate);
 
         me.mon(me.getStore(), 'datachanged', function () {
+            me.store.filter();
             set_button_status();
         });
     },
@@ -42718,10 +43004,92 @@
             };
         });
 
+        let entryeditor = Proxmox.UserName === 'root@pam' ? 'PVE.lxc.EntryEdit' : undefined;
+
+        PVE.Utils.forEachLxcEntry(function(i, confid) {
+            rows[confid] = {
+                group: 8,
+                order: i,
+                tdCls: 'pve-itype-icon-pci',
+                editor: entryeditor,
+                header: gettext('Entry') + ' (' + confid + ')',
+            };
+        });
+
         var baseurl = 'nodes/' + nodename + '/lxc/' + vmid + '/config';
 
         me.selModel = Ext.create('Ext.selection.RowModel', {});
 
+        var createAddMenu = function() {
+            return new Ext.menu.Menu({
+                listeners: {
+                    beforeshow: function(menu) {
+                        Proxmox.Utils.API2Request({
+                            url: `/api2/extjs/nodes/${nodename}/lxc/${vmid}/config`,
+                            method: 'GET',
+                            success: function(response) {
+                                let ostype = response.result.data.ostype;
+                                let mountEntry = menu.items.find(item => item.text === gettext('Mount Entry'));
+                                let devicePassthrough = menu.items.find(item => item.text === gettext('Device Passthrough'));
+                                if (mountEntry && devicePassthrough) {
+                                    mountEntry.setHidden(ostype !== 'oci');
+                                    devicePassthrough.setHidden(ostype === 'oci');
+                                }
+                            },
+                        });
+                    }
+                },
+                items: [
+                    {
+                        text: gettext('Mount Point'),
+                        iconCls: 'fa fa-fw fa-hdd-o black',
+                        disabled: !caps.vms['VM.Config.Disk'],
+                        handler: function() {
+                            Ext.create('PVE.lxc.MountPointEdit', {
+                                autoShow: true,
+                                url: `/api2/extjs/${baseurl}`,
+                                unprivileged: me.getObjectValue('unprivileged'),
+                                pveSelNode: me.pveSelNode,
+                                listeners: {
+                                    destroy: () => me.reload(),
+                                },
+                            });
+                        },
+                    },
+                    {
+                        text: gettext('Mount Entry'),
+                        iconCls: 'pve-itype-icon-pci',
+                        disabled: Proxmox.UserName !== 'root@pam',
+                        handler: function() {
+                            Ext.create('PVE.lxc.EntryEdit', {
+                                autoShow: true,
+                                url: `/api2/extjs/${baseurl}`,
+                                pveSelNode: me.pveSelNode,
+                                listeners: {
+                                    destroy: () => me.reload(),
+                                },
+                            });
+                        },
+                    },
+                    {
+                        text: gettext('Device Passthrough'),
+                        iconCls: 'pve-itype-icon-pci',
+                        disabled: Proxmox.UserName !== 'root@pam',
+                        handler: function() {
+                            Ext.create('PVE.lxc.DeviceEdit', {
+                                autoShow: true,
+                                url: `/api2/extjs/${baseurl}`,
+                                pveSelNode: me.pveSelNode,
+                                listeners: {
+                                    destroy: () => me.reload(),
+                                },
+                            });
+                        },
+                    },
+                ],
+            });
+        };
+
         var run_resize = function () {
             var rec = me.selModel.getSelection()[0];
             if (!rec) {
@@ -42893,6 +43261,7 @@
             let isUnusedDisk = key.match(/^unused\d+/);
             let isUsedDisk = isDisk && !isUnusedDisk;
             let isDevice = key.match(/^dev\d+/);
+            let isEntry = key.match(/^entry\d+/);
 
             let noedit = isDelete || !rowdef.editor;
             if (!noedit && Proxmox.UserName !== 'root@pam' && key.match(/^mp\d+$/)) {
@@ -42908,7 +43277,7 @@
             reassign_menuitem.setDisabled(isRootFS);
             resize_menuitem.setDisabled(isUnusedDisk);
 
-            remove_btn.setDisabled(!(isDisk || isDevice) || isRootFS || !diskCap || pending);
+            remove_btn.setDisabled(!(isDisk || isDevice || isEntry) || isRootFS || !diskCap || pending);
             revert_btn.setDisabled(!pending);
 
             remove_btn.setText(isUsedDisk ? remove_btn.altText : remove_btn.defaultText);
@@ -42947,41 +43316,7 @@
             tbar: [
                 {
                     text: gettext('Add'),
-                    menu: new Ext.menu.Menu({
-                        items: [
-                            {
-                                text: gettext('Mount Point'),
-                                iconCls: 'fa fa-fw fa-hdd-o black',
-                                disabled: !caps.vms['VM.Config.Disk'],
-                                handler: function () {
-                                    Ext.create('PVE.lxc.MountPointEdit', {
-                                        autoShow: true,
-                                        url: `/api2/extjs/${baseurl}`,
-                                        unprivileged: me.getObjectValue('unprivileged'),
-                                        pveSelNode: me.pveSelNode,
-                                        listeners: {
-                                            destroy: () => me.reload(),
-                                        },
-                                    });
-                                },
-                            },
-                            {
-                                text: gettext('Device Passthrough'),
-                                iconCls: 'pve-itype-icon-pci',
-                                disabled: Proxmox.UserName !== 'root@pam',
-                                handler: function () {
-                                    Ext.create('PVE.lxc.DeviceEdit', {
-                                        autoShow: true,
-                                        url: `/api2/extjs/${baseurl}`,
-                                        pveSelNode: me.pveSelNode,
-                                        listeners: {
-                                            destroy: () => me.reload(),
-                                        },
-                                    });
-                                },
-                            },
-                        ],
-                    }),
+                    menu: createAddMenu(),
                 },
                 edit_btn,
                 remove_btn,
